packages/podman-extension-api/src/podman-extension-api.d.ts

```diff
@@ -0,0 +1,23 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { RunOptions, RunResult } from '@podman-desktop/api';
+
+export interface PodmanExtensionApi {
+  exec(args?: string[], options?: RunOptions): Promise<RunResult>;
```

Comment:
Not sure the `RunOptions` of the @podman-desktop/api are a good match here ?

https://github.com/containers/podman-desktop/blob/4f9e876667fe5ab48d26bd9d2d1ab728e3a89dd5/packages/extension-api/src/extension-api.d.ts#L4033

The isAdmin bother me

Reply:
> Not sure the RunOptions of the @podman-desktop/api are a good match here ?

It is a good first iteration

> The isAdmin bother me

Some commands if you run them require admin privileges. it does not mean you getting them right away, it just force PD to request user to login.